crates/provider/src/provider/with_block.rs

```diff
@@ -109,7 +109,7 @@ where
             // append the block id to the params
             if let serde_json::Value::Array(ref mut arr) = ser {
                 arr.push(block_id);
-            } else if let serde_json::Value::Null = ser {
+            } else if ser == serde_json::Value::Null {
```

Comment:
can condense to `is_null()` while we're cleaning :)

Reply:
fixed